service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -756,6 +757,26 @@ fi
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASHRC}"
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASH_PROFILE}"
 
+####################################
+# Run a user provided startup script
+####################################
+
+# If the user has provided a startup script, run it after workbench setup,
+# but before restarting Jupyter and running tests.
+
+readonly USER_STARTUP_SCRIPT="$(get_metadata_value "instance/attributes/terra-user-startup-script")"
+if [[ -n "${USER_STARTUP_SCRIPT}" ]]; then
+  emit "Running user provided startup script..."
+  readonly USER_STARTUP_SCRIPT_FILE="${USER_TERRA_CONFIG_DIR}/user-startup-script.sh"
+
+  # Copy the user's startup script to the user's .terra directory
+  curl -L "${USER_STARTUP_SCRIPT}" > "${USER_STARTUP_SCRIPT_FILE}"
+
+  # Run the user's startup script as the login user
+  chmod +x "${USER_STARTUP_SCRIPT_FILE}"
+  source "${USER_STARTUP_SCRIPT_FILE}"
```

Comment:
how are you running it as login user?

Reply:
oh ty, the comment should be "run as root user".

Reply:
I thought it was intentional that we give root access? This is the current behavior:
```
User jupyter may run the following commands on vm-name:
    (ALL : ALL) ALL
    (ALL : ALL) NOPASSWD: ALL
```

Reply:
interesting! didn't realize we allow sudo there. 

so it's not a change in behavior, sgtm